src/Infrastructure.VS.UnitTests/SolutionInfoProviderTests.cs

```diff
@@ -64,19 +77,64 @@ public async Task Get_ReturnsExpectedValue(string solutionNameToReturn)
         }
 
         [TestMethod]
-        public async Task Get_ServiceCalledOnBackgroundThread()
+        [DataRow(null)]
+        [DataRow("a value")]
+        public void Get_ReturnsExpectedValue(string solutionNameToReturn)
+        {
+            var solution = CreateIVsSolution(solutionNameToReturn);
+            var serviceProvider = CreateServiceProviderWithSolution(solution.Object);
+
+            var testSubject = CreateTestSubject(serviceProvider.Object);
+
+            var actual = testSubject.GetFullSolutionFilePath();
+            actual.Should().Be(solutionNameToReturn);
+        }
+
+        [TestMethod]
+        public async Task GetAsync_ServiceCalledOnUIThread()
         {
             var calls = new List<string>();
 
             var solution = CreateIVsSolution("any", () => calls.Add("GetSolutionName"));
             var serviceProvider = CreateServiceProviderWithSolution(solution.Object, () => calls.Add("GetService"));
-            var threadHandling = CreateThreadHandlingWithRunOnUICallback(() => calls.Add("RunOnUIThread"));
```

Comment:
I removed the `CreateThreadHandling...` helper method since it was only used in once place, and we need another version for the synchronous test since it calls different thread handling method.